src/test/cloudWatchLogs/registry/logStreamRegistry.test.ts

```diff
@@ -150,21 +150,62 @@ describe('LogStreamRegistry', async function () {
             )
         })
 
-        it('adds stream id to searches from searchLogGroup, but not viewLogStream', function () {
-            const viewLogStreamContent = registry.getLogContent(registeredUri)
-            const searchLogGroupContent = registry.getLogContent(searchLogGroupUri)
-
-            assert(viewLogStreamContent)
-            assert(searchLogGroupContent)
-
-            const searchLogGroupStreamName = parseCloudWatchLogsUri(searchLogGroupUri).parameters.streamName
-            const viewLogStreamName = parseCloudWatchLogsUri(registeredUri).parameters.streamName
-
-            assert(searchLogGroupStreamName)
-            assert(viewLogStreamName)
+        it('registers stream ids to map', function () {
+            registry.getLogContent(searchLogGroupUri) // We run this to create the mappings
+            const streamIDMap = registry.getStreamIdMap(searchLogGroupUri)
+            const expectedMap = new Map<number, string>([
+                [0, testStreamNames[0]],
+                [1, testStreamNames[1]],
+            ])
+            assert.deepStrictEqual(streamIDMap, expectedMap)
```

Comment:
This test should also assert that the streamIdMap is disposed / deleted after the document is closed.

Reply:
In order to do so I would have to programmatically close the `vscode.TextDocument` which I am not sure if it is supported. (https://github.com/microsoft/vscode/issues/49740 asks about this a few years ago and don't see anything more recent in linked to issue) Is there a workaround/another way to test this?

Reply:
It's not supported. When you run into these kinds of situations you can write your logic in such a way to make it testable by abstracting the untestable parts. For example, expose a way to deregister a URI and then test that. The document close event would call this deregister function but your test code is agnostic to that fact.